scrapy/downloadermiddlewares/httpproxy.py

```diff
@@ -9,11 +9,13 @@
 
 from scrapy.utils.httpobj import urlparse_cached
 from scrapy.exceptions import NotConfigured
+from scrapy.utils.python import to_bytes
 
 
 class HttpProxyMiddleware(object):
 
-    def __init__(self):
+    def __init__(self, auth_encoding=None):
```

Comment:
Hm, are you suggesting users can create instances of this middleware themselves? Tests is not a good use case, but there may be subclasses. In this case having `latin1` decoding may be the best option indeed.